store/pg_engine.go

```diff
@@ -227,91 +217,146 @@ func (db *DB) seedFile(name string) error {
+		latestSchemaPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestSchemaFile)
+		buf, err := migrationFS.ReadFile(latestSchemaPath)
+		if err != nil {
+			return fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
+		}
+		latestDataSchemaPath := fmt.Sprintf("migration/%s/%s", db.schemaVersion.String(), latestDataFile)
+		dataBuf, err := migrationFS.ReadFile(latestDataSchemaPath)
+		if err != nil {
+			return fmt.Errorf("failed to read latest schema %q, error %w", latestSchemaPath, err)
+		}
+		// We will create the database together with initial schema and data migration.
+		stmt := fmt.Sprintf("CREATE DATABASE %s;\n\\connect \"%s\";\n%s\n%s", databaseName, databaseName, buf, dataBuf)
 		if _, _, err := d.ExecuteMigration(
 			ctx,
 			&dbdriver.MigrationInfo{
-				ReleaseVersion: db.serverVersion,
-				Version:        createDatabaseSchemaVersion,
-				Namespace:      databaseName,
-				Database:       databaseName,
-				Environment:    "", /* unused in execute migration */
-				Source:         dbdriver.LIBRARY,
-				Type:           dbdriver.Baseline,
-				Description:    fmt.Sprintf("Create database %s.", databaseName),
-				CreateDatabase: true,
+				ReleaseVersion:        db.serverVersion,
+				UseSemanticVersion:    true,
+				Version:               db.schemaVersion.String(),
+				SemanticVersionSuffix: time.Now().Format("20060102150405"),
+				Namespace:             databaseName,
+				Database:              databaseName,
+				Environment:           "", /* unused in execute migration */
+				Source:                dbdriver.LIBRARY,
+				Type:                  dbdriver.Baseline,
+				Description:           fmt.Sprintf("Migrate %s.", latestSchemaPath),
+				CreateDatabase:        true,
 			},
-			createDatabaseStatement,
+			stmt,
 		); err != nil {
-			return fmt.Errorf("failed to migrate create database schema, error: %v", err)
+			return fmt.Errorf("failed to migrate initial schema version %q, error: %v", latestSchemaPath, err)
 		}
+		db.l.Info("Completed database initial migration.")
+		return nil
 	}
 
 	// Apply migrations
-	names, err := fs.Glob(migrationFS, "migration/*.sql")
+	versionNames, err := fs.Glob(migrationFS, "migration/*")
 	if err != nil {
 		return err
 	}
-	// Sort the migration up file in ascending order.
-	sort.Strings(names)
+	var versions []semver.Version
+	for _, name := range versionNames {
+		v, err := semver.Make(strings.TrimPrefix(name, "migration/"))
+		if err != nil {
+			return fmt.Errorf("invalid migration file path %q, error %w", name, err)
+		}
+		versions = append(versions, v)
+	}
+	// Sort the migration semantic version in ascending order.
+	sort.Slice(versions, func(i, j int) bool {
+		return versions[i].LT(versions[j])
+	})
 
-	maxVer := versionFromInt(db.schemaVersion)
-	for _, name := range names {
-		versionPrefix := strings.Split(filepath.Base(name), "__")[0]
-		version, err := strconv.Atoi(versionPrefix)
+	for _, version := range versions {
```

Comment:
I would suggest to extract a method to returned the versions to be migrated (including L300 ~ L307)

And we can write a test to check the if the to-be migrated versions are expected.